contracts/src/Teleporter/upgrades/TeleporterUpgradeable.sol

```diff
@@ -217,6 +221,33 @@ abstract contract TeleporterUpgradeable is Context, ITeleporterReceiver, Reentra
      */
     function _checkTeleporterUpgradeAccess() internal virtual;
 
+    /**
+     * @dev Sends a cross chain message using the TeleporterMessenger contract.
+     * @return `messageID` The unique identifier for the Teleporter message.
+     * @return `adjustedFeeAmount` The adjusted fee amount after transferring the fee token.
+     */
+    function _sendTeleporterMessage(TeleporterMessageInput memory messageInput)
```

Comment:
Could you add unit tests for this new function?

Reply:
added